Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2887,6 +2887,7 @@ protected void onPostExecute(ServerResponse serverResponse) {
                                 Log.i("BranchSDK", "Branch API Error: Conflicting resource error code from API");
                                 handleFailure(0, status);
                             }
+                            requestQueue_.remove(thisReq_);
```

Comment:
Thank you. Absolutely correct use case. Added changes to remove the request before calling the callback